src/files/XDRFile.cpp

```diff
@@ -41,6 +45,26 @@ void XDRFile::write_opaque(const char* data, uint32_t count) {
     write_char(filler.data(), num_filler);
 }
 
+void XDRFile::read_gmx_long_opaque(std::vector<char>& data) {
+    const uint64_t count = read_single_u64();
+    if (count > std::numeric_limits<size_t>::max()) {
+        // count would overflow the address space
+        throw file_error("compressed data is too large to be allocated");
+    }
```

Comment:
Gromacs needs to put [more complicated](https://gitlab.com/gromacs/gromacs/-/blob/5ee3606d1ca3ce4d1c0767c0b0b4a31c5715bf09/src/gromacs/fileio/libxdrf.cpp#L796-810) code to make sure that the batch size does not overflow the positive range of `int`.
Luckily, chemfiles `BinaryFile` handles everything with `size_t`. The only issue would occur when reading data which cannot be addressed by `size_t`. This code produces a human friendly error in these cases.